packages/core/components/AutoField/fields/SelectField/index.tsx

```diff
@@ -27,19 +28,10 @@ export const SelectField = ({
     >
       <select
         id={id}
+        title={label || name}
         className={getClassName("input")}
         disabled={readOnly}
-        onChange={(e) => {
-          if (
-            e.currentTarget.value === "true" ||
-            e.currentTarget.value === "false"
-          ) {
-            onChange(JSON.parse(e.currentTarget.value));
-            return;
-          }
-
-          onChange(e.currentTarget.value);
-        }}
+        onChange={({ target: { value } }) => safeJsonParse(value) ? onChange(JSON.parse(value)) : onChange(value)}
```

Comment:
Can you use the return value of `safeJsonParse` so we're not running JSON parse twice?